datafusion/expr-common/src/signature.rs

```diff
@@ -93,60 +96,89 @@ pub enum Volatility {
 pub enum TypeSignature {
     /// One or more arguments of a common type out of a list of valid types.
     ///
+    /// For functions that take no arguments (e.g. `random()` see [`TypeSignature::Nullary`]).
+    ///
     /// # Examples
-    /// A function such as `concat` is `Variadic(vec![DataType::Utf8, DataType::LargeUtf8])`
+    ///
+    /// A function such as `concat` is `Variadic(vec![DataType::Utf8,
+    /// DataType::LargeUtf8])`
```

Comment:
nit:
Maybe it would be more readable to go without `DataType:` for type names?

Reply:
in https://github.com/apache/datafusion/pull/13840